File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.java
class Solution {
    public int[] compute(int n,int column){
        int y = n%column;
        int x = 0;
        // int v = 0;
        while (n>=column){
            n-=column;
            x+=1;
        }
        return new int[]{x,y};
    }
    public boolean searchMatrix(int[][] mat, int key) {
        int column = mat[0].length;
        int low = 0;
        int high = mat.length*column - 1;
        int mid,x,y;
        while(low<=high){
            mid = (low+high)/2;
            int[] v = compute(mid,column);
            x = v[0];
            y = v[1];
            if (mat[x][y]==key){
                return true;
            }
            else if(mat[x][y]>key){
                high = mid-1;
            }
            else{
                low = mid+1;
            }
        }
        return false;
        
    }
}

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m = len(matrix)
        n = len(matrix[0])
        
        def giveMatrixCoordinate(i):
            # 0 = 0,0
            # 3 = 0,3
            # 6 = 1,2
            x = i//n
            y = i%n
            return [x,y]
        # print(0,giveMatrixCoordinate(0))
        # print(3,giveMatrixCoordinate(3))
        # print(6,giveMatrixCoordinate(6))
        # print(10,giveMatrixCoordinate(10))
        low = 0
        high= (m*n)-1
        while low<=high:
            mid = (low+high)//2
            x,y = giveMatrixCoordinate(mid)
            if matrix[x][y]==target:
                return True
            elif target < matrix[x][y]:
                high = mid-1
            else:
                low = mid+1
        return False